components/form/UploadImageInput.tsx

```diff
@@ -8,20 +8,24 @@ type UploadImageProps = {
   setImageSrc: (src: string) => void;
   setError?: (error: string) => void;
   isRequired?: boolean;
+  imageType: 'item' | 'profile';
 };
 
-const CDN =
-  'https://undfcbmldjkujposixvn.supabase.co/storage/v1/object/public/images/';
+const CDN = `${process.env.NEXT_PUBLIC_SUPABASE_URL}/storage/v1/object/public/images/`;
 
 /**
  * @description this function will add any image to our project storage.
+ * @param setImageSrc is a required parameter that accepts a string. It will set the value of the image source if the upload is successful.
+ * @param setError is an optional parameter that accepts a string. It will set the value of the error message if the image upload fails.
  * @param isRequired is an optional parameter that accepts a boolean. It will consider the value of false unless explicitly assigned the value of true.
+ * @param imageType is a required parameter that accepts either 'item' or 'profile'. It will determine the type of image to be uploaded in the imageFileUpload function.
  */
 
 const UploadImageInput: React.FC<UploadImageProps> = ({
   setImageSrc,
   setError,
   isRequired: isRequired = false,
+  imageType,
 }) => {
   const supabase = createClientComponentClient();
```

Comment:
Hey @nataliiazab , 
   Alphonso asked me to give your PR, a quick once over, and although you didn't right this bit I was wondering if you could refactor this quickly? Instead a creating a new supabase client like this we are always trying to use the function newClient() in the supabase/utils folder. 
   This is because this function first checks if there's an instance of this client so we don't create more then one as it was causing some error (this is called the singleton principle and a fun pattern to learn about). So if could update this for us, that would be amazing!

Reply:
Hi @nichgalzin, thanks for reviewing my PR! 

❌ I keep receiving the following error when I change from createClientComponentClient() to newClient().
> ![image](https://github.com/user-attachments/assets/85095050-b966-4833-96ae-6083d3b63bc5)

I've been adjusting the bucket permission policies, but haven't made any progress. Could you please check if you're experiencing the same error? ☺️ 

Here is one of the changes I've tried:
> ![image](https://github.com/user-attachments/assets/e7af655e-8359-4561-9ebf-2a0947bbb211)

Reply:
Hey @nataliiazab,
    Sorry for the slow reply on this. I'm having a play and not getting either version to work. I'll message you on discord, maybe we could hop on a call sometime this week and try to resolve things together?

Reply:
@nichgalzin - I've added the changes we discussed. Thanks a million for your help and review ⭐